tests/events/network_client/main.c

```diff
@@ -0,0 +1,200 @@
+#include <limits.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <stdio.h>
+
+#include "events.h"
+#include "parsenum.h"
+#include "sock.h"
+#include "warnp.h"
+
+static int done;
+
+static int
+poke(struct sock_addr ** sas)
+{
+	int client_sock;
+
+	/* Attempt to connect, but don't complain about any failures. */
+	if ((client_sock = sock_connect(sas)) == -1)
```

Comment:
For this test, it would be ideal if `sock_connect()` could return the reason in errno.  For example, if we get `ECONNREFUSED` from `connect()` then we know that the server shut itself down (as the test is designed to do), whereas anything else would indicate an actual error.

However, I don't think it's worth messing with `sock_connect()` just for this test.

Reply:
"Return the reason in errno" is a bit ill-defined.  If we have 2 IP addresses, and one returns ECONNREFUSED and the other returns EHOSTUNREACH, which should we return?

Reply:
Yes, that's another good reason to avoid messing with `sock_connect`.